### PR TITLE
Client-side GitHub API calls

### LINKED ISSUE #2400: Netlify builds failing on fork PRs
**Describe the bug**

Recently we're seeing failed builds on contributor PRs. It's related to the Github API but unlike the rate limit issues in #2160, we're seeing 401 forbidden status on the Github API (same as #1472), which results in breaking builds.

Netlify removes environment variables from the builds of external contributors, e.g. from build logs
```
12:09:49 PM: The following environment variables have been removed from this build to keep them secret:
12:09:49 PM: - ALGOLIA_ADMIN_KEY
12:09:49 PM: - GITHUB_TOKEN_READ_ONLY_PROD
12:09:49 PM: - GATSBY_ALGOLIA_SEARCH_KEY
12:09:49 PM: - GITHUB_TOKEN_READ_ONLY_DEV
12:09:49 PM: - CROWDIN_API_KEY
12:09:49 PM: - GITHUB_TOKEN_READ_ONLY_STAGING
12:09:49 PM: - CAPTCHA_SECRET
12:09:49 PM: - DEFI_PULSE_API_KEY
12:09:49 PM: - FAUCET_PRIVATE_KEY
12:09:49 PM: - GITHUB_TOKEN
12:09:49 PM: - ETHERSCAN_API_KEY
```

**To Reproduce**

See recent contributor PRs, e.g https://github.com/ethereum/ethereum-org-website/pull/2397#partial-pull-merging ([build logs](https://app.netlify.com/sites/ethereumorg/deploys/60215524dd4b9b0008d75a44)) & https://github.com/ethereum/ethereum-org-website/pull/2398 ([build logs](https://app.netlify.com/sites/ethereumorg/deploys/60219a8ce705a80008878698)).

**Expected behavior**

Builds should pass 😄

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Pulls GitHub contributor information client side, for the tutorial and developer documents pages. 
- Should prevent hitting GitHub API rate limits which have been resulting in many failed Netlify builds.

## Related Issue
Fixes #2400

cc: @samajammin 🎉?

### CODE REVIEW

**Comment by wackerow** [comment]
@samajammin Thoughts on giving this a try? Switched it over to Apollo so it's using GraphQL instead of REST. Seems to be running smoothly on Netlify and locally.

File: src/apollo.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    : process.env.CONTEXT === "deploy-preview"
+    ? process.env.GITHUB_TOKEN_READ_ONLY_STAGING
+    : process.env.GITHUB_TOKEN_READ_ONLY_DEV
+
+const client = new ApolloClient({
+  link: new HttpLink({
+    uri: `https://api.github.com/graphql`,
+    fetch,
+    headers: {
+      Authorization: `Bearer ${GITHUB_TOKEN_READ_ONLY}`,
```
Comment (samajammin) [inline src/apollo.js:16]: I believe we need to update this to a `GATSBY_...` ENV variable so it's accessible from the fronted. See my original PR here: https://github.com/ethereum/ethereum-org-website/pull/2296/files#diff-9bd41b9894ddb2787f97e601c440c1cf7b1018f8a52498fbec796bdc422efcf4

**Review by samajammin — CHANGES_REQUESTED** [review]
See my comment. Currently we're getting 401 unauthorized response, I believe because the `GITHUB_TOKEN_READ_ONLY` variable is not accessible from the client:

Image 2021-04-01 at 5 39 40 PM

**Comment by samajammin** [comment]
Another thought - with the upgrade to Gatsby v3 (#2528), Gatsby add incremental builds by default:
https://www.gatsbyjs.com/docs/reference/release-notes/v3.0/#incremental-builds-in-oss

I'm curious if this will fix our API woes. I'm not clear on exactly how it works - would cache be preserved in Netlify for PRs into `dev`? Or does it build any new branch/fork PR from scratch?

**Comment by wackerow** [comment]
@samajammin hm, still getting 401 with `GATSBY_GITHUB_TOKEN_READ_ONLY`... 
I'm not sure if Netlify would be able to share cache from another branch to its deploy preview builds. We'll find out soon though since Gatsby v3 changes were merged in... we'll see how new PRs do on builds.

**Comment by samajammin** [comment]
@wackerow sorry I wasn't clear. We'd also need to add that `GATSBY_GITHUB_TOKEN_READ_ONLY` env variable for it to work 😄  - I've added it to Netlify & just ran a rebuild.

**Comment by samajammin** [comment]
More context on the `GATSBY_` prefix for client side variables: https://www.gatsbyjs.com/docs/how-to/local-development/environment-variables/#client-side-javascript

**Comment by wackerow** [comment]
@samajammin beautiful, looks like it's passing after 18 min. May be worth bringing this in and seeing how it affects our builds

**Comment by samajammin** [comment]
Keep getting build failures in Netlify around the GH API (see #2802):
```
1:57:11 PM: Url path: /developers/local-environment/
1:57:11 PM: Plugin: none
1:57:11 PM: 
1:57:11 PM: 
1:57:11 PM:   FetchError: request to https://api.github.com/graphql failed, reason: Client n  etwork socket disconnected before secure TLS connection was established
1:57:11 PM:   
1:57:11 PM:   - index.js:1461 ClientRequest.<anonymous>
1:57:11 PM:     [repo]/[node-fetch]/lib/index.js:1461:11
1:57:11 PM:   
1:57:11 PM:   - destroy.js:92 emitErrorNT
1:57:11 PM:     internal/streams/destroy.js:92:8
1:57:11 PM:   
1:57:11 PM:   - destroy.js:60 emitErrorAndCloseNT
1:57:11 PM:     internal/streams/destroy.js:60:3
1:57:11 PM:   
1:57:11 PM:   - task_queues.js:84 processTicksAndRejections
1:57:11 PM:     internal/process/task_queues.js:84:21
1:57:11 PM:   
1:57:11 PM: 
1:57:11 PM: not finished run page queries - 2124.630s
1:57:11 PM: error Command failed with exit code 1.
```

Testing out removing those queries on build: https://github.com/ethereum/ethereum-org-website/pull/2713/commits/a7353f19eb1f951792f2b1ed4d3d6f90c9b545a6

We'll see if that fixes things...

**Pull request merged**